web-app/HeimScripts/heatmap/limmaUtils.R

```diff
@@ -33,3 +29,11 @@ getDesign <- function(measurements) {
 getSubset1Length <- function(measurements) {
   sum(grepl(pattern = SUBSET1REGEX, x = colnames(measurements))) # returns number of column names satisfying regexp.
 }
+
+# to check if row  contains at least one row that contains 3 non missing values
+rowContainsAtLeastThreeData <- function (row) {sum(!is.na(row)) > 2}
+
```

Comment:
it is a valid measurement if there is at least one row for which these _two_ criteria hold:
* at least 3 non NA measurements
* at least 1 non NA measurement per subset
please apply the second constraint too